java/AxivityReader.java

```diff
@@ -312,14 +322,16 @@ private static void readCwaBuffer(ByteBuffer buf,
                     x = xRaw / 256.0;
                     y = yRaw / 256.0;
                     z = zRaw / 256.0;
-                    timeValue = getEpochMillis(blockTime);
+                    t = zonedWithDSTCorrection(blockTime).toInstant().toEpochMilli();
 
-                    epochWriter.newValues(timeValue, x, y, z, temperature, errCounter);
+                    epochWriter.newValues(t, x, y, z, temperature, errCounter);
 
                 }
             } catch (Exception excep) {
                 excep.printStackTrace(System.err);
-                System.err.println("block err @ " + lastBlockTime[0].toString() + ": " + excep.toString());
```

Comment:
For printing purposes, zone and DST-correct the time. Note that internally the code still works with `LocalDateTime` objects because of some low-level processes.

Reply:
This makes me think that it would be better to conduct timeZone corrections at the EpochWriter.java level.

Reply:
Perhaps it would be good to perform the correction at the very source instead of delegating to subsequent processes? Maybe EpochWriter.java should just receive the values all corrected and be in charge of just writing/formatting.
Also, I think `lastBlockTime` would still need to be DST-corrected for printing anyway.

Reply:
After further reflection, I agree with you @chanshing